pkg/pulumiyaml/run.go

```diff
@@ -1861,6 +1864,101 @@ func (e *programEvaluator) evaluateBuiltinSecret(s *ast.SecretExpr) (interface{}
 	return pulumi.ToSecret(expr), true
 }
 
+func (e *programEvaluator) evaluateInterpolatedBuiltinAssetArchive(x, s ast.Expr) (interface{}, bool) {
+	_, isConstant := s.(*ast.StringExpr)
+	v, b := e.evaluateExpr(s)
+	if !b {
+		return nil, false
+	}
+
+	createAssetArchiveF := e.lift(func(args ...interface{}) (interface{}, bool) {
+		value, ok := args[0].(string)
+		if !ok {
+			return e.error(s, fmt.Sprintf("Argument to Fn::ReadFile must be a string, got %v", reflect.TypeOf(args[0])))
+		}
+
+		switch x.(type) {
+		case *ast.StringAssetExpr:
+			return pulumi.NewStringAsset(value), true
+		case *ast.FileArchiveExpr:
+			path, err := e.sanitizePath(value, isConstant)
+			if err != nil {
+				return e.error(s, err.Error())
+			}
+			return pulumi.NewFileArchive(path), true
+		case *ast.FileAssetExpr:
+			path, err := e.sanitizePath(value, isConstant)
+			if err != nil {
+				return e.error(s, err.Error())
+			}
+			return pulumi.NewFileAsset(path), true
+		case *ast.RemoteArchiveExpr:
+			if !isConstant {
+				return e.error(s, "Argument to Fn::RemoteArchiveExpr must be a constantr")
+			}
+			return pulumi.NewRemoteArchive(value), true
+		case *ast.RemoteAssetExpr:
+			if !isConstant {
+				return e.error(s, "Argument to Fn::RemoteAssetExpr must be a constant")
+			}
+			return pulumi.NewRemoteAsset(value), true
+
+		}
+		return e.error(s, "unhandled expression")
+	})
+
+	return createAssetArchiveF(v)
+}
+
+func (e *programEvaluator) sanitizePath(value string, isConstant bool) (string, error) {
```

Comment:
I think `path` would be a much cleaner name then `value`.
```suggestion
func (e *programEvaluator) sanitizePath(path string, isConstant bool) (string, error) {
```